### PR TITLE
fix can not export service bug

### PR DESCRIPTION
**Pull request by adohe-zz**
Please refer #22247 for more details.
@bgrant0607 I just add a quick&simple fix, would you mind reviewing this?

### CODE REVIEW

**Comment by AdoHe** [comment]
The Jenkins unit/integration failed for strange reason:
```
ERROR: Step 鈥楶ublish JUnit test result report鈥� failed: No test report files were found. Configuration error?
```

File: pkg/api/rest/rest.go [hunk 1]
```diff
@@ -221,6 +221,7 @@ type StandardStorage interface {
 	GracefulDeleter
 	CollectionDeleter
 	Watcher
+	Exporter
```
Comment (smarterclayton) [inline pkg/api/rest/rest.go]: I don't think this should be required - it's desirable, but not required.

**Comment by AdoHe** [comment]
@smarterclayton just update this. Besides now the export output of service is:
```
NAME         CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes                <none>        443/TCP   <unknown>
```
I am thinking whether we should update `CLUSTER-IP` to `<none>` instead of empty.

**Comment by bgrant0607** [comment]
@AdoHe Yes, I'd prefer to output `<none>` rather than leave it empty.

**Comment by AdoHe** [comment]
@bgrant0607 I think about this for a while, and I prefer to use `<unknown>` instead of `<none>`

**Comment by bgrant0607** [comment]
@AdoHe Good idea. `<unknown>` SGTM.

File: pkg/registry/service/registry.go [hunk 2]
```diff
@@ -30,21 +33,22 @@ type Registry interface {
 	DeleteService(ctx api.Context, name string) error
 	UpdateService(ctx api.Context, svc *api.Service) (*api.Service, error)
 	WatchServices(ctx api.Context, options *api.ListOptions) (watch.Interface, error)
+	ExportService(ctx api.Context, name string, options unversioned.ExportOptions) (*api.Service, error)
 }
 
 // storage puts strong typing around storage calls
 type storage struct {
-	rest.StandardStorage
+	ss rest.StandardStorage
```
Comment (lavalamp) [inline pkg/registry/service/registry.go]: Why did you name this?

Comment (lavalamp) [inline pkg/registry/service/registry.go]: Please leave unnamed, or if you have to name it, use a name like 'delegate' or something else more descriptive. Tiny names are good for local variable names in short functions, but not good as members in structs.

File: pkg/registry/service/registry.go [hunk 3]
```diff
@@ -81,7 +85,19 @@ func (s *storage) UpdateService(ctx api.Context, svc *api.Service) (*api.Service
 }
 
 func (s *storage) WatchServices(ctx api.Context, options *api.ListOptions) (watch.Interface, error) {
-	return s.Watch(ctx, options)
+	return s.ss.Watch(ctx, options)
+}
+
+func (s *storage) ExportService(ctx api.Context, name string, options unversioned.ExportOptions) (*api.Service, error) {
+	exporter, isExporter := s.ss.(rest.Exporter)
```
Comment (lavalamp) [inline pkg/registry/service/registry.go]: If you added the name to get this line to work, the way to specify it without the name change is `s.StandardStorage.(rest.Exporter)`

**Comment by lavalamp** [comment]
Minor comments.

**Comment by lavalamp** [comment]
LGTM when rebased, thanks

File: pkg/registry/service/strategy.go [hunk 4]
```diff
@@ -90,7 +90,7 @@ func (svcStrategy) Export(obj runtime.Object, exact bool) error {
 		return nil
 	}
 	if t.Spec.ClusterIP != api.ClusterIPNone {
-		t.Spec.ClusterIP = ""
+		t.Spec.ClusterIP = "<unknown>"
```
Comment (kargakis) [inline pkg/registry/service/strategy.go]: Why this change?

Comment (smarterclayton) [inline pkg/registry/service/strategy.go]: This prevents the exported resource from being created on another cluster, which defeats the purpose of export.

Comment (mkumatag) [inline pkg/registry/service/strategy.go]: facing issue -  https://github.com/kubernetes/kubernetes/issues/47976, https://github.com/kubernetes/kubernetes/issues/28548
